switches/http/http.cpp

```diff
@@ -0,0 +1,1280 @@
+            finishResponse(req, 200, ""); // or perhaps {"return_value":null,"cmd":"sendEvent","name":nodeName}?
+            return;
+        }
+    }
+    
+    // Handler: Subscribe to an event stream
+    
+    void HttpInterface::evSubscribe(HttpRequest* req, strings& args)
+    {
+        // eventSubscriptions[conn] is an unordered set of strings
+        if (args.size() == 1)
+            eventSubscriptions[req].insert("*");
+        else
+            for (strings::iterator i = args.begin()+1; i != args.end(); ++i)
+                eventSubscriptions[req].insert(*i);
+        
+        strings headers;
+        headers.push_back("Content-Type: text/event-stream");
+        headers.push_back("Cache-Control: no-cache");
+        headers.push_back("Connection: keep-alive");
+        addHeaders(req, headers);
+        appendResponse(req,200,true,"");
+        // connection must stay open!
+    }
+    
+    // Handler: Compile and store a program into the node, and remember it for introspection
+    
+    void HttpInterface::evLoad(HttpRequest* req, strings& args)
+    {
+        if (verbose)
+            cerr << "PUT /nodes/" << args[0].c_str() << " trying to load aesl script\n";
+        const char* buffer = req->content.c_str();
+        size_t pos = req->content.find("file=");
+        if (pos != std::string::npos)
+        {
+            aeslLoadMemory(buffer+pos+5, req->content.size()-pos-5);
+            finishResponse(req, 200, "");
+        }
+        else
+            finishResponse(req, 400, "");
+    }
+    
+    // Handler: Reset nodes and rerun
+    
+    void HttpInterface::evReset(HttpRequest* req, strings& args)
+    {
+        for (NodesDescriptionsMap::iterator descIt = nodesDescriptions.begin();
+             descIt != nodesDescriptions.end(); ++descIt)
+        {
+            bool ok;
+            nodeId = getNodeId(descIt->second.name, 0, &ok);
+            if (!ok)
+                continue;
+            char nodeName[128];
+            wstringtocstr(descIt->second.name, nodeName);
+            
+            //this->lock(); // inherited from Dashel::Hub
+            
+            Reset(nodeId).serialize(asebaStream); // reset node
+            asebaStream->flush();
+            Run(nodeId).serialize(asebaStream);   // re-run node
+            asebaStream->flush();
+            if (strcmp(nodeName, "thymio-II") == 0)
+            {
+                strings args;
+                args.push_back("motor.left.target");
+                args.push_back("0");
+                sendSetVariable(nodeName, args);
+                args[0] = "motor.right.target";
+                sendSetVariable(nodeName, args);
+            }
+            size_t eventPos;
+            if (commonDefinitions.events.contains(UTF8ToWString("reset"), &eventPos))
+            {
+                strings data;
+                data.push_back("reset");
+                sendEvent(nodeName,data);
+            }
+            
+            this->unlock();
```

Comment:
Locking isn't working for me on Windows and ought to be removed here until it does.